code/game/objects/items/weapons/cosmetics.dm

```diff
@@ -491,3 +491,71 @@
 		return
 	if (prob(25))
 		shatter()
+
+/obj/item/weapon/nanitecontacts
+	name = "Nanite Contacts"
+	desc = "Deploys nanobots to your eyes to change their color."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "nanite_contact"
+	flags = FPRINT
+	w_class = W_CLASS_TINY
+	var/color_r = 255
+	var/color_g = 255
+	var/color_b = 255
+
+/obj/item/weapon/nanitecontacts/New()
+	..()
+	color_r = rand(0,255)
+	color_g = rand(0,255)
+	color_b = rand(0,255)
+	update_icon()
+
+/obj/item/weapon/nanitecontacts/update_icon()
+	overlays.len = 0
+	var/icon/contact_color = new/icon("icon" = 'icons/obj/items.dmi', "icon_state" = "contacts_overlay")
```

Comment:
Make this create a new `image()` with the icon, set its color variable and add that as overlay instead.

`/icon` operations are ludicrously expensive and they're not necessary here.

Reply:
how to do this

Reply:
also help with checking if user is alive and still has contacts in their hand

Reply:
Basically you want to do

```dm
var/image/I = image(icon = 'icons/obj/items.dmi', icon_state = "contacts_overlay")
I.color = rgb(color_r, color_g, color_b)
overlays += I
```
Might not produce the same blending as with an icon but eh.